src/networking.c

```diff
@@ -1717,65 +1723,68 @@ void readQueryFromClient(aeEventLoop *el, int fd, void *privdata, int mask) {
      * the event loop. This is the case if threaded I/O is enabled. */
     if (postponeClientRead(c)) return;
 
-    readlen = PROTO_IOBUF_LEN;
-    /* If this is a multi bulk request, and we are processing a bulk reply
-     * that is large enough, try to maximize the probability that the query
-     * buffer contains exactly the SDS string representing the object, even
-     * at the risk of requiring more read(2) calls. This way the function
-     * processMultiBulkBuffer() can avoid copying buffers to create the
-     * Redis Object representing the argument. */
-    if (c->reqtype == PROTO_REQ_MULTIBULK && c->multibulklen && c->bulklen != -1
-        && c->bulklen >= PROTO_MBULK_BIG_ARG)
+    for (;;)    // read from socket until EAGAIN is returned
```

Comment:
This is of particular importance.  We need to ensure we fully flush all buffers before returning as epoll only checks the kernel's buffer.